syft/messaging/plan/plan.py

```diff
@@ -578,6 +580,100 @@ def detail(worker: AbstractWorker, plan_tuple: tuple) -> "Plan":
 
         return plan
 
+    @staticmethod
+    def bufferize(worker: AbstractWorker, plan: "Plan") -> PlanPB:
+        """
+        This function takes the attributes of a Plan and saves them in a Protobuf message
+        Args:
+            worker (AbstractWorker): the worker doing the serialization
+            plan (Plan): a Plan object
+        Returns:
+            PlanPB: a Protobuf message holding the unique attributes of the Plan object
+
+        """
+        protobuf_plan = PlanPB()
+
+        sy.serde.protobuf.proto.set_protobuf_id(protobuf_plan.id, plan.id)
+
+        protobuf_operations = [
+            sy.serde.protobuf.serde._bufferize(worker, operation).operation
+            for operation in plan.operations
+        ]
+        protobuf_plan.operations.extend(protobuf_operations)
+
+        protobuf_plan.state.CopyFrom(sy.serde.protobuf.serde._bufferize(worker, plan.state))
+
+        protobuf_plan.include_state = plan.include_state
+        protobuf_plan.is_built = plan.is_built
+        protobuf_plan.name = plan.name
+        protobuf_plan.tags.extend(plan.tags)
+
+        if protobuf_plan.description:
+            protobuf_plan.description = plan.description
+
+        if type(plan.placeholders) == type(dict()):
+            placeholders = plan.placeholders.values()
+        else:
+            placeholders = plan.placeholders
```

Comment:
I'm not 100% sure that `plan.placeholders` can be anything other than a `dict` here. `replace_with_placeholders()` seems to have multiple potential return types, but I'm not sure they actually get stored anywhere. Possible this `if/else` can be simplified to just assume that it will receive `dict`.

Reply:
@LaRiffle Is the structure of `plan.placeholders` always a `dict` now?